### PR TITLE
[ISSUE #8601]When isPopShouldStop hit,unlock queueLockManager

### LINKED ISSUE #8601: [Bug] when isPopShouldStop hit，queueLockManager will not release rightly
### Before Creating the Bug Report

- [X] I found a bug, not just asking a question, which should be created in [GitHub Discussions](https://github.com/apache/rocketmq/discussions).

- [X] I have searched the [GitHub Issues](https://github.com/apache/rocketmq/issues) and [GitHub Discussions](https://github.com/apache/rocketmq/discussions)  of this repository and believe that this is not a duplicate.

- [X] I have confirmed that this bug belongs to the current repository, not other repositories of RocketMQ.

### Runtime platform environment

any

### RocketMQ version

develop

### JDK Version

_No response_

### Describe the Bug

whern isPopShouldStop hit, we should release the lock

### Steps to Reproduce

review the code

### What Did You Expect to See?

review the code

### What Did You See Instead?

review the code

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Which Issue(s) This PR Fixes

Fixes #8601

### Brief Description

When isPopShouldStop hit，unlock queueLockManager

### How Did You Test This Change?

review the code

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java [hunk 1]
```diff
@@ -543,6 +543,7 @@ private CompletableFuture<Long> popMsgFromQueue(String topic, String attemptId,
         if (isPopShouldStop(topic, requestHeader.getConsumerGroup(), queueId)) {
             POP_LOGGER.warn("Too much msgs unacked, then stop poping. topic={}, group={}, queueId={}", topic, requestHeader.getConsumerGroup(), queueId);
             restNum = this.brokerController.getMessageStore().getMaxOffsetInQueue(topic, queueId) - offset + restNum;
+            future.whenComplete((result, throwable) -> queueLockManager.unLock(lockKey));
```
Comment (lizhanhui) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java:546]: Moving line#552 before line#543 looks better.

Comment (leizhiyuan) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java]: ok

**Review by lizhanhui — DISMISSED** [review]
Good catch

**Review by cserwen — DISMISSED** [review]

**Review by lizhimins — APPROVED** [review]
Good catch

**Pull request merged**